<?xml version='1.0' encoding='utf-8'?>
<launch>
	<arg name="debug" default="false" />
	<arg name="gui" default="true" />
	<arg name="pause" default="false" />
	<arg name="world" default="$(find diff_drive_package)/worlds/floor_plan.world"/>

	<!-- Launching the world file -->
	<include file="$(find gazebo_ros)/launch/empty_world.launch">
		<arg name="world_name" value="$(arg world)"/>
		<arg name="debug" value="$(arg debug)" />
		<arg name="gui" value="$(arg gui)" />
		<arg name="paused" value="$(arg pause)" />
		<arg name="use_sim_time" value="true" />
	</include>

	<!-- Goal point -->
	<rosparam file="$(find diff_drive_package)/config/test_params.yaml" />

	<!-- Turtlebot navigation node -->
	<!-- <node name="turtlebot_navigate" pkg="diff_drive_package" type="turtlebot_navigate" /> -->
	<!-- <node name="turtlebot_navigate" pkg="diff_drive_package" type="turtlebot_navigate">
  		<rosparam file="$(find diff_drive_package)/config/test_params.yaml" />
	</node> -->


	<!-- Launching the diff drive robot-->
	<arg name="model" default="burger"/>
	<arg name="rob_x" default="6.0"/>
	<arg name="rob_y" default="4.0"/>
	<arg name="rob_z" default="0.0"/>

	<param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
	<node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model) -x $(arg rob_x) -y $(arg rob_y) -z $(arg rob_z) -param robot_description" />
</launch>
